src/http.ts

```diff
@@ -43,84 +158,100 @@ export async function loadSourceImage ({ cacheDir, url, requestEtag, modifiers,
   const cacheKey = String(murmurhash(url))
   const inputCacheFile = join(fileCache, cacheKey)
 
-  const headers = new Headers(requestHeaders)
-  let sourceMetadata: SourceMetadata | undefined
-  if (existsSync(inputCacheFile)) {
-    sourceMetadata = (await metadataStore.getItem(`source:${cacheKey}`)) as
+  await markUsageStart(metadataStore, cacheKey, inputCacheFile)
+  await maybePruneCache(metadataStore)
+
+  function finalize () {
+    return markUsageComplete(metadataStore, cacheKey)
+  }
```

Comment:
All those functions are async and it might cause consistency problems sometimes if `handler` does run concurrently.

To address this some kind of mutex is needed or we add queue to handler to ensure only one image transformation runs at the same time.

Alternatively we accept that this is not perfect and call it good enough

Reply:
Added lock in https://github.com/netlify/netlify-ipx/pull/136/commits/7a0693e6f4f8c0f9c55e85b584988cf62edbdf8b